### PR TITLE
Generalize hostinfo results structure and organize code into packages

### PR DESCRIPTION
**Pull request by itzg**

### CODE REVIEW

File: utils/timestamp.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package utils
+
+import "time"
+
+type NowTimestampMillisFunc func() int64
+
+var NowTimestampMillis NowTimestampMillisFunc = func() int64 {
```
Comment (itzg) [inline utils/timestamp.go]: Not sure if this is idiomatic go, but I did this to enable "injecting" a mock timestamp retrieval for unit tests to get a deterministic timestamp

File: types/session.go [hunk 2]
```diff
@@ -157,12 +158,15 @@ func (s *Session) handleFrame(f *FrameMsg) {
...
 func (s *Session) handleHostInfo(f *FrameMsg) {
-	if hinfo := NewHostInfo(f); hinfo != nil {
-		go func(s *Session, hinfo HostInfo, f *FrameMsg) {
+	if hinfo := hostinfo.NewHostInfo(*f.GetRawParams()); hinfo != nil {
+		go func(s *Session, hinfo hostinfo.HostInfo, f *FrameMsg) {
 			cr, err := hinfo.Run()
 			if err != nil {
 			} else {
-				s.Send(NewHostInfoResponse(f, hinfo, cr))
+				response := &HostInfoResponse{}
```
Comment (itzg) [inline types/session.go]: Good catch. That was a lingering refactor attempt.

File: utils/timestamp.go (hunk 1 above)
Comment (rphillips) [inline utils/timestamp.go]: great idea

**Comment by rphillips** [comment]
+1

**Pull request merged**